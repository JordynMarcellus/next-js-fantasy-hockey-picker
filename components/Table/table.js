import React from "react";
import Checkbox from "../checkbox";
import * as tableStyles from "./table.styles";

/*

  data structure: 
  
  {Columns} : [{label: "Corsica rank", key: "corsica-rank", cellChildren: (<tableStyles.RankingAvatar>
      <span>
        {player.position}
        {player.rank}
      </span>
    </RankingAvatar>}), {
      header: "Player name"
    }]

  {Column.header: "Corsica rank"}
  {Column.key="corsica-rank"}
  {Column.cellChildren=(
    <tableStyles.RankingAvatar>
      <span>
        {player.position}
        {player.rank}
      </span>
    </RankingAvatar>
    )
  }
//  <tableStyles.StyledCell as="th" textAlign="center">
*/

const table = props => (
  <tableStyles.StyledTable>
    <tableStyles.StyledTableHeader>
      <tr>
        {props.columns.map(tableColumn => {
          return (
            <tableStyles.StyledCell
              key={`table-header--row--${tableColumn.componentKey}`}
              as="th"
              textAlign="center">
              {tableColumn.label}
            </tableStyles.StyledCell>
          );
        })}
      </tr>
    </tableStyles.StyledTableHeader>
    <tbody>
      {props.players.map(player => {
        const onPlayerSelect = playerId => value => {
          props.selectPlayer({ value, playerId });
        };
        return (
          <tableStyles.StyledRow
            key={player.id}
            onClick={() => {
              props.handlePlayerDrilldown(player.id);
            }}
            isDrafted={player.selected}>
            <tableStyles.StyledCell>
              <tableStyles.RankingAvatar position={player.position}>
                <span>
                  {player.position}
                  {player.rank}
                </span>
              </tableStyles.RankingAvatar>
            </tableStyles.StyledCell>
            <tableStyles.StyledCell textTransform={"capitalize"}>
              {player.name}
            </tableStyles.StyledCell>
            <tableStyles.StyledCell>{player.team}</tableStyles.StyledCell>
            <tableStyles.StyledCell textAlign="right">
              {player.rating}
            </tableStyles.StyledCell>
            <tableStyles.StyledCell>
              <Checkbox
                isSelected={player.selected}
                onChange={onPlayerSelect(player.id)}
              />
            </tableStyles.StyledCell>
          </tableStyles.StyledRow>
        );
      })}
    </tbody>
  </tableStyles.StyledTable>
);

export default table;
