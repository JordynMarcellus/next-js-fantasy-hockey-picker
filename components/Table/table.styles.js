import React from "react";
import styled from "@emotion/styled";

const positionMap = {
  c: "#3867d6",
  lw: "#eb3b5a",
  rw: "#20bf6b",
  ld: "#8854d0",
  rd: "#8854d0",
};

export const StyledTable = styled.table`
  border-collapse: collapse;
  margin: 0 auto;
  background-color: #fff;
`;

export const StyledTableHeader = styled.thead`
  border-bottom: 2px solid #333;
`;

export const StyledRow = styled.tr`
  background-color: ${({ isDrafted }) => (isDrafted ? "red" : "#fff")};
`;

export const StyledCell = styled.td(
  props => `
    text-align: ${props.textAlign}};
    padding: 0.5rem;
  `
);

StyledCell.defaultProps = {
  textAlign: "left",
};

export const RankingAvatar = styled.div`
  background-color: ${({ position }) => positionMap[position.toLowerCase()]};
  color: white;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
`;
