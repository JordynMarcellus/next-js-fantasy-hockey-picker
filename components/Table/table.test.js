import "@testing-library/jest-dom/extend-expect";
import { render, fireEvent } from "@testing-library/react";
import Table from "./table";
const testData = [
  {
    id: "d23e3f7a-7cd7-4496-8af4-6551cef3b94f",
    name: "nikita kucherov",
    selected: false,
    position: "RW",
    team: "TB",
    rank: 1,
    rating: "88.63",
  },
  {
    id: "6073ce66-15f0-430c-a665-caac2f3f715f",
    name: "brad marchand",
    selected: false,
    position: "LW",
    team: "BOS",
    rank: 1,
    rating: "86.16",
  },
  {
    id: "6af98394-5469-413c-b0f4-b1edf1710697",
    name: "nathan mackinnon",
    selected: false,
    position: "C",
    team: "COL",
    rank: 2,
    rating: "86.10",
  },
];

describe("Table component", () => {
  it("renders a table with playerData", () => {
    const columns = [
      {
        header: "Corsica rank",
        key: "corsica-rank",
      },
      {
        header: "Player name",
        key: "player-name",
      },
      {
        header: "Current team",
        key: "current-team",
      },
      {
        header: "Corsica rating",
        key: "corsica-rating",
      },
      {
        header: "Drafted?",
        key: "drafted",
      },
    ];
    const { container, debug } = render(
      <Table columns={columns} players={testData} selectPlayer={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
