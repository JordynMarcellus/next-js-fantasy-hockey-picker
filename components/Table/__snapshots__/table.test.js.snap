// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component renders a table with playerData 1`] = `
.emotion-54 {
  border-collapse: collapse;
  margin: 0 auto;
  background-color: #fff;
}

.emotion-10 {
  border-bottom: 2px solid #333;
}

.emotion-0 {
  text-align: center;
  padding: 0.5rem;
}

.emotion-24 {
  background-color: #fff;
}

.emotion-14 {
  text-align: left;
  padding: 0.5rem;
}

.emotion-12 {
  background-color: #20bf6b;
  color: white;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-20 {
  text-align: right;
  padding: 0.5rem;
}

.emotion-26 {
  background-color: #eb3b5a;
  color: white;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-40 {
  background-color: #3867d6;
  color: white;
  border-radius: 50%;
  width: 50px;
  height: 50px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <table
    class="emotion-54 emotion-55"
  >
    <thead
      class="emotion-10 emotion-11"
    >
      <tr>
        <th
          class="emotion-0 emotion-1"
        />
        <th
          class="emotion-0 emotion-1"
        />
        <th
          class="emotion-0 emotion-1"
        />
        <th
          class="emotion-0 emotion-1"
        />
        <th
          class="emotion-0 emotion-1"
        />
      </tr>
    </thead>
    <tbody>
      <tr
        class="emotion-24 emotion-25"
      >
        <td
          class="emotion-14 emotion-1"
        >
          <div
            class="emotion-12 emotion-13"
          >
            <span>
              RW
              1
            </span>
          </div>
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          nikita kucherov
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          TB
        </td>
        <td
          class="emotion-20 emotion-1"
        >
          88.63
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr
        class="emotion-24 emotion-25"
      >
        <td
          class="emotion-14 emotion-1"
        >
          <div
            class="emotion-26 emotion-13"
          >
            <span>
              LW
              1
            </span>
          </div>
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          brad marchand
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          BOS
        </td>
        <td
          class="emotion-20 emotion-1"
        >
          86.16
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          <input
            type="checkbox"
          />
        </td>
      </tr>
      <tr
        class="emotion-24 emotion-25"
      >
        <td
          class="emotion-14 emotion-1"
        >
          <div
            class="emotion-40 emotion-13"
          >
            <span>
              C
              2
            </span>
          </div>
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          nathan mackinnon
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          COL
        </td>
        <td
          class="emotion-20 emotion-1"
        >
          86.10
        </td>
        <td
          class="emotion-14 emotion-1"
        >
          <input
            type="checkbox"
          />
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
